src/components/SearchForm.js

```diff
@@ -1,35 +1,37 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
 
 const SearchForm = ({onFormChange}) => {
-  const [searchTerm, setSearchTerm] = useState("");
-  const [characterCategory, setCharacterCategory] = useState("");
-  const [characterRarity, setCharacterRarity] = useState("");
-  const [characterType, setCharacterType] = useState("");
-  const [isUserDeck, setIsUserDeck] = useState(false);
-
   return (
     <div className="flex flex-row flex-wrap justify-around mx-5">
       {/* //search field */}
       <form
-        onSubmit={(e) => e.preventDefault()}
-        onChange={(e) => onFormChange(new FormData(e.currentTarget))}
+
+        // calls onFormChange with formData. Schema:
+        // {
+        //   searchTerm: string (empty string if not present),
+        //   characterCategory: string (empty string if ALL)
+        //   characterType: string (empty string if ALL)
+        //   characterRarity: string (empty string if ALL)
+        //   isUserDeck: true or else this key is not present
+        // }
+        onChange={(e) => {
+          const formData = Object.fromEntries(new FormData(e.currentTarget));
```

Comment:
Simplified getting the formData out by managing in each input. Described the schema of "formData" in the above comment